### PR TITLE
Merge for gsoc

### PR DESCRIPTION
**Pull request by Rishabh42**
- Disabled fbturbo

- deleted Physics.xo and turtleblocks.xo

- Aded lines to download physics & turtleblocks during the build process

- Updated readme

### CODE REVIEW

File: README.md
Comment (quozl) [inline README.md]: - [ ] _version_ could be _fork_ or _branch_,
- [ ] mention of _Raspberry pi 2 & 3_ is superfluous, since it is already the scope set by previous paragraph,
- [ ] you do not say how to run the build process with a template; perhaps refer to the templates section.

File: bootstrap.d/11-mod.sh
Comment (quozl) [inline bootstrap.d/11-mod.sh:39]: - [ ] no reason given for removing this blank line, perhaps it wasn't intentional, can't tell..

File: files/etc/skel/.xsession
Comment (quozl) [inline files/etc/skel/.xsession]: - [x] no explanation given for removing this, there is bound to be some good reason.

File: rpi23-gen-image.sh
Comment (quozl) [inline rpi23-gen-image.sh]: - [ ] changing the comment to be inconsistent with the code inside the conditional ... is wrong.

**Review by quozl — CHANGES_REQUESTED** [review]
- [ ] please rebase your branch before making a pull request, thanks.  See https://developer.sugarlabs.org/contributing.md.html#keep%20your%20pull-request%20up%20to%20date and https://developer.sugarlabs.org/contributing.md.html#keep%20your%20fork%20up%20to%20date

File: files/etc/skel/.xsession
Comment (Rishabh42) [reply files/etc/skel/.xsession]: I had earlier merged SL's code with my branch which merged a couple of old changes with my fork, so I had to revert that PR and cherry pick the changes, I guess this got deleted as a consequence. Wil add it again

File: rpi23-gen-image.sh
Comment (Rishabh42) [reply rpi23-gen-image.sh]: Should I revert back to the original line which was inconsistent with the code?

File: README.md
Comment (Rishabh42) [reply README.md]: Okay will make the proposed changes

**Comment by quozl** [comment]
On Sun, Aug 27, 2017 at 01:49:26AM -0700, Rishabh Thaney wrote:
> I had earlier merged SL's code with my branch which merged a couple
> of old changes with my fork, so I had to revert that PR and cherry
> pick the changes, I guess this got deleted as a consequence. Wil add
> it again

Why would you do that?  Is it needed now after your earlier changes
were merged?

**Comment by quozl** [comment]
On Sun, Aug 27, 2017 at 08:50:11AM +0000, Rishabh Thaney wrote:
> Should I revert back to the original line which was inconsistent
> with the code?

No, it wasn't inconsistent upstream, you made it inconsistent with
your change to the commands inside the conditional block.

File: bootstrap.d/11-mod.sh
Comment (Rishabh42) [reply bootstrap.d/11-mod.sh:39]: I don't think an extra blank line would affect the way this script executes, I can add it if it's necessary though.

**Comment by Rishabh42** [comment]
Should we use the 'rebase and merge' option from the drop down list next to the 'merge pull request' option to merge the PR instead of manually rebasing using the terminal?

**Comment by quozl** [comment]
Sqash and merge, rebase and merge, these are not the problem.  The main problem is _you are making changes without explaining them properly in commit messages attached to the change_,

You did ask for review.

Please read again [Making Commits](https://developer.sugarlabs.org/contributing.md.html#making%20commits).  Note the ways in which your commits have not followed Sugar Labs contribution guidelines.  As soon as I hit something I don't understand in your changes, I've gone looking for the explanation in the commit message and been disappointed.

Please review GitHub's collapsed view of the commits by clicking on the [files changed](https://github.com/sugarlabs/rpi23-gen-image/pull/13/files) tab of the pull request.  Note how there are several changes in that view which are not explained in the commit messages.

You should try rebasing, and for safety do so in a separate branch, assuming your default remote is this repository;

```bash
git checkout -b gsoc-rebase
git fetch
git rebase --interactive
```

Whether you rebase into your own [repository on GitHub](https://github.com/rishabh42/rpi23-gen-image/)  or your own repository on a computer or phone, doesn't matter.  I don't see why you haven't done that except that you haven't got any mentor who can explain it to you well, or you haven't spent the time to learn git.

I'll move comments around a bit in this pull request in a moment, and respond to your latest commit b85e1ca.

File: rpi23-gen-image.sh
Comment (quozl) [reply rpi23-gen-image.sh]: No, it wasn't inconsistent upstream, you made it inconsistent with your change to the commands inside the conditional block.

File: files/etc/skel/.xsession
Comment (quozl) [reply files/etc/skel/.xsession]: Why would you do that? Is it needed now after your earlier changes were merged?

File: bootstrap.d/11-mod.sh
Comment (quozl) [reply bootstrap.d/11-mod.sh:39]: Nothing to do with the way the script executes; it will just take a little bit longer with a blank line than without, but the amount of time is microscopic.  My review comment is that you've removed a blank line, you didn't say you'd done it, so it makes it look like you don't know what you are changing.  Changes unrelated to the purpose of a commit make `git bisect` very hard to use, and I'm a heavy user of `git bisect`, so I'm sensitive to unrelated changes.  If you can justify the change as necessary (e.g. an extra blank line was added by accident in a previous commit), then you can also explain the change in the commit description.

File: rpi23-gen-image.sh
Comment (quozl) [inline rpi23-gen-image.sh:353]: - [ ] changing the comment to be inconsistent with the code inside the conditional ... is wrong.

Since I've already raised this in a previous review, and your response in b85e1ca9 did nothing about it, I'll explain further.

In your file `rpi23-gen-image.sh` is a comment "Don't clobber an old build", above a conditional check for the existence of the build directory, which executes `rm -rf` to delete it.  This is just simply inconsistent.  The comment doesn't say what the code does.  The comment is a lie.  We can't abide lies in code.

**Review by quozl — CHANGES_REQUESTED** [review]

**Comment by Rishabh42** [comment]
Thanks for informing me that I should explain each change I make, will keep that in mind while making changes going forward.

File: rpi23-gen-image.sh
Comment (Rishabh42) [reply rpi23-gen-image.sh:353]: I think I get your point now, will make some changes and get back to you.

File: bootstrap.d/11-mod.sh
Comment (Rishabh42) [reply bootstrap.d/11-mod.sh:39]: I understand your concern, it was added accidently in the previous commit, I realised this while making changes to the file.

File: rpi23-gen-image.sh
Comment (Rishabh42) [reply rpi23-gen-image.sh:353]: Should I change the comment to: "To check for the existence of an old build folder and delete it if it does"?

Comment (quozl) [reply rpi23-gen-image.sh:353]: You seem to be missing the history of change in this code.  You're thrashing around like a dying fish.

Originally upstream wrote this to require that you delete builds yourself.  This is a useful feature because it prevents losing valuable work.  It is easy to overcome.

In 2dd011a18b6932f6229f78fac8e449c11526344f and 646006f4604e3adf4962d2ec2d8006cb464b88e6 you decided to delete the build automatically; but you didn't change the comment.

My review at the time was based on the assumption that you were doing this change for yourself only (not me or @Hrishi1999), and _rather than change rpi23-gen-image.sh to clobber old builds, you might instead learn to use compound statements in shell, the ctrl+r key for recalling them, or small wrapper scripts,_  You chose not to respond to that review comment.

These changes eventually became a squash and rebase commit e52bb817a22a3e782798346c2839207e8dc8b57c which made the comment inconsistent with the code.

In response, I raised #5, rather than insist.

In c2ea0feb8161aa2a2d94b19e650b0c89e659f081 you changed the comment to _To delete an old build_.

Your pull request now seeks to change the comment from _To delete an old build_ to _Don't clobber an old build_.

And now you are asking if the comment should be changed again ... it is truly disappointing.

Please use the tools available.  Read the changes proposed by your pull request; the sections in red and green.  That's what anyone else will be reading; and if you don't understand why a change is proposed, perhaps you need to fix it before making the pull request.

Comment (Rishabh42) [reply rpi23-gen-image.sh:353]: Sorry to disappoint you due to misconception on my side. So are you suggesting that I change that code and the comment to what was initially there in the upstream? Which won't delete the build folder automatically but will notify the user that it exists and it won't proceed further with building the image because of that.

Comment (quozl) [reply rpi23-gen-image.sh:353]: > are you suggesting that I change that code and the comment to what was initially there in the upstream?

I'm saying in this review, that _changing the comment to be inconsistent with the code inside the conditional_, which is what your pull request is proposing to do, _is wrong_.

Can you see that your pull request is asking for this change?

That's blocking this pull request.

The rest of what I said is illustrative of your misunderstanding.

It is up to you what to propose change, via commit to the `gsoc` branch assoicated with this pull request, but the change at the moment is wrong because it makes the comment lie about the code.  I've also said weeks ago your previous changes to this code are wrong as well, but that's not directly what this pull request is about.

You've also used "git push" to rewrite the history of the Sugar Labs repository, merging in all your changes to the `master` branch without review.  The last two commits used to be 667c8b080ab057444176d7717881a3d77082f25a and e52bb817a22a3e782798346c2839207e8dc8b57c but everything you did in your private repository is now merged with the history here.  This means that when someone clones the repository, they get a copy of every one of those activity bundle files which are then deleted, because they form part of history now.  What a waste.

Comment (Rishabh42) [reply rpi23-gen-image.sh:353]: Alright, I've resetted that code to what was originally there in the upstream for now while I figure out a better solution for this.
Regarding merging changes to master, that happened by accident sorry about that. 
I'll go through the previous changes too and fix them within a couple of days.
I hope this PR can be merged now

**Pull request merged**
